src/Raven.Server/Documents/Versioning/VersioningStorage.cs

```diff
@@ -286,43 +286,43 @@ private VersioningConfigurationCollection GetVersioningConfiguration(CollectionN
                 if (configuration == null)
                     configuration = GetVersioningConfiguration(collectionName);
 
-                DeleteOldRevisions(context, table, lowerId, configuration.MaxRevisions, nonPersistentFlags);
+                DeleteOldRevisions(context, table, lowerId, configuration, nonPersistentFlags);
             }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice lowerId, 
-            long? maxRevisions, NonPersistentDocumentFlags nonPersistentFlags)
+            VersioningConfigurationCollection configuration, NonPersistentDocumentFlags nonPersistentFlags)
         {
             using (GetKeyPrefix(context, lowerId, out Slice prefixSlice))
             {
                 // We delete the old revisions after we put the current one, 
                 // because in case that MaxRevisions is 3 or lower we may get a revision document from replication
                 // which is old. But because we put it first, we make sure to clean this document, because of the order to the revisions.
                 var revisionsCount = IncrementCountOfRevisions(context, prefixSlice, 1);
-                DeleteOldRevisions(context, table, prefixSlice, maxRevisions, revisionsCount, nonPersistentFlags);
+                DeleteOldRevisions(context, table, prefixSlice, configuration, revisionsCount, nonPersistentFlags);
             }
         }
 
-        private void DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, long? maxRevisions, long revisionsCount, 
+        private void DeleteOldRevisions(DocumentsOperationContext context, Table table, Slice prefixSlice, VersioningConfigurationCollection configuration, long revisionsCount, 
             NonPersistentDocumentFlags nonPersistentFlags)
         {
             if ((nonPersistentFlags & NonPersistentDocumentFlags.FromSmuggler) == NonPersistentDocumentFlags.FromSmuggler)
                 return;
 
-            if (maxRevisions.HasValue == false || maxRevisions.Value == int.MaxValue)
+            if (configuration.MaxRevisions.HasValue == false || configuration.MaxRevisions.Value == int.MaxValue)
```

Comment:
Not sure that I agree. What if I just want to delete all revisions earlier than 2 weeks?
Don't care about the number of them.